### PR TITLE
Sim1h + deadlocks fixed + app-spec update + ...

### PR DESCRIPTION
**Pull request by lucksus**
## PR summary

This is an aggregated PR that includes several parts that in concert constitute a big shift in terms of usability and reliability.

* Adds sim1h as new networking implementation
* Replaces n3h by sim1h in app-spec tests
* Migrates app-spec to a new Try-o-rama version
* Revamps our use of thread synchronisation primitives and fixes several potential deadlocks 
* Adds `UUID` to DNA configs, which also deactivates the DNA hash check if set (https://github.com/holochain/holochain-rust/pull/1724, fixes https://github.com/holochain/holochain-rust/issues/1600)

## testing/benchmarking notes

Running app-spec locally with sim1h can be done through the new nix-shell command `hc-app-spec-test-sim1h`. It is configured though to assume a dynamodb instance running on `http://localhost:8000`. The new nix-shell command `dynamodb-memory` starts an in-memory instance on that port. 

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

Please check one of the following, relating to the [CHANGELOG-UNRELEASED.md](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md)

- [x] this is a code change that effects some consumer (e.g. zome developers) of holochain core so it is added to the CHANGELOG-UNRELEASED.md (linked above), with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: add-aws-env.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# environment variables used by rust tests directly
+export AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE
+export AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+# config file used by aws cli tool
+export AWS_CONFIG_FILE=`pwd`/.aws/config
+RUST_LOG=sim1h=trace
```
Comment (lucksus) [inline add-aws-env.sh]: I don't think we need this file any more as I copied over the nix commands from sim1h. Right, @maackle?

File: app_spec/test/files/crypto.js [hunk 2]
```diff
@@ -1,101 +1,109 @@
...
-    scenario('encrypt_and_decrypt_message', async (s, t, { alice, bob }) => {
+
+    scenario('encrypt_and_decrypt_message', async (s, t) => {
+      const { alice, bob } = await s.players({alice: one, bob: one}, true)
       const message = "Hello everyone! Time to start the secret meeting";
-    
-      const EncryptResult = await bob.app.call("simple", "encrypt", { payload:message });
-      
+
+      const EncryptResult = await bob.call('app', "simple", "encrypt", { payload:message });
```
Comment (lucksus) [inline app_spec/test/files/crypto.js]: I guess these changes are due to a Try-o-rama update, @maackle?

File: cli/src/cli/js-tests-scaffold/index.js [hunk 3]
```diff
@@ -1,7 +1,17 @@
+
+
+///////////////////////////////////
+///
+///
+///            TODO
+///
+///
+////////////////////////////////////
```
Comment (lucksus) [inline cli/src/cli/js-tests-scaffold/index.js]: ?

Comment (lucksus) [inline cli/src/cli/js-tests-scaffold/index.js]: ?

File: conductor_api/src/conductor/admin.rs [hunk 4]
```diff
@@ -209,12 +209,11 @@ impl ConductorAdmin for Conductor {
                     .into(),
             },
         };
-        new_config.instances.push(new_instance_config);
-        new_config.check_consistency(&mut self.dna_loader)?;
-        let instance = self.instantiate_from_config(id, Some(&mut new_config))?;
+        self.config.instances.push(new_instance_config);
+        self.config.check_consistency(&mut self.dna_loader)?;
```
Comment (lucksus) [inline conductor_api/src/conductor/admin.rs]: See my comment on the other PR about these lines, @maackle: https://github.com/holochain/holochain-rust/pull/1724#discussion_r330476516

**Comment by willemolding** [comment]
Does this need a rebase or something? There can't possibly be 100 file changes to add sim1h!

**Pull request merged**